app/controllers/registration_controller.rb

```diff
@@ -0,0 +1,30 @@
+# RegistrationController
+#
+# Handles incoming registration requests from tool
+# consumers.
+class RegistrationController < ApplicationController
+  skip_before_filter :verify_authenticity_token, only: :register
+  include RegistrationHelper
+
+  def register
+    # Fetch the a custom tool consumer profile
+    tcp_helper = ToolConsumerProfile.new(URI.parse(params[:tc_profile_url]), jwt_access_token)
+
+    # Do failure redirect to tool consumer if missing required
+    redirect_to registration_failure_url('Missing required capabilities') and return unless tcp_helper.supports_required_capabilities?
+
+    # Create a new tool proxy
+    tool_proxy = ToolProxy.new(tcp_url: tcp_helper.tcp_url, base_url: request.base_url)
```

Comment:
Sounds great. I just took a look and from what I can tell the tool proxy registration service only does the create request using the reg_key/reg_password rather than a token in the Authorization header. Is that correct?